### ISSUE TITLE
get_dataset_names implemenation/documentation is inconsistent

### ISSUE THREAD

**Issue by nathanfranklin** [simphony/simphony-common#218]
#200 implemented in #206 added the method `get_dataset_names` 

the method is documented as follows in `ABCModelingEngine`:
```
def get_dataset_names(self):
	""" Returns the names of the all the datasets in the engine workspace.
```

But the testing framework, implementation in H5CUDS and in the UML diagrams treats the method as returns an iterator.

**Comment by nathanfranklin** [comment]
I think we should switch the implementation and documentation so that `get_dataset_names` returns a list of names.

**Comment by nathanfranklin** [comment]
Closed by #230
